modular_zubbers/modules/plexagon_selfserve/code/plexagon_selfserve.dm

```diff
@@ -0,0 +1,282 @@
+
+/datum/computer_file/program/crew_self_serve/kill_program(mob/user)
+	computer.crew_manifest_update = FALSE
+	if(!isnull(authenticated_card))
+		authenticated_card = null
+
+	return ..()
+
+/datum/computer_file/program/crew_self_serve/ui_act(action, params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+
+	authenticate(computer.computer_id_slot)
+	switch(action)
+		if("PRG_change_status")
+			if(off_duty_check())
+				if (!authenticated_card)
+					return;
+
+				if(!(clock_in()))
+					return
+				log_admin("[key_name(usr)] clocked in as \an [authenticated_card.assignment].")
+
+				var/datum/mind/user_mind = usr.mind
+				if(user_mind)
+					user_mind.clocked_out_of_job = FALSE
+
+			else
+				if (!authenticated_card)
+					return;
+
+				log_admin("[key_name(usr)] clocked out as \an [authenticated_card.assignment].")
+				clock_out()
+				var/mob/living/carbon/human/human_user = usr
+				if(human_user)
+					var/obj/item/storage/lockbox/timeclock/shame_box = new /obj/item/storage/lockbox/timeclock(src, authenticated_card)
+					human_user.secure_items(eligible_items = SELF_SERVE_RETURN_ITEMS, incoming_box = shame_box)
+
+				var/datum/mind/user_mind = usr.mind
+				if(user_mind)
+					user_mind.clocked_out_of_job = TRUE
+
+				if(job_is_CMD_or_SEC())
+					message_admins("[key_name(usr)] has clocked out as a [authenticated_card.assignment]. [ADMIN_JMP(authenticated_card)]")
+
+				computer.update_static_data_for_all_viewers()
+				playsound(computer, 'sound/machines/ping.ogg', 50, FALSE)
+			return TRUE
+		if("PRG_eject_id")
+			eject_inserted_id(usr)
+			return TRUE
+
+
+/datum/computer_file/program/crew_self_serve/ui_data(mob/user)
+	var/list/data = list()
+	data["authCard"] = authenticated_card ? authenticated_card.name : "-----"
+	data["authCardTimeLocked"] = id_cooldown_check()
+
+	return data
+
+/datum/computer_file/program/crew_self_serve/ui_static_data(mob/user)
+	var/list/data = list()
+	data["stationAlertLevel"] = SSsecurity_level.get_current_level_as_text()
+	if(authenticated_card)
+		data["authIDName"] = authenticated_card.registered_name ? authenticated_card.registered_name : "-----"
+		data["authIDRank"] = authenticated_card.assignment ? authenticated_card.assignment : "Unassigned"
+		data["authCardHOPLocked"] = id_locked_check()
+		data["trimClockedOut"] = off_duty_check()
+		if(authenticated_card.trim)
+			var/datum/id_trim/card_trim = authenticated_card.trim
+			data["trimAssignment"] = card_trim.assignment ? card_trim.assignment : ""
+		else
+			data["trimAssignment"] = ""
+	else
+		data["authIDName"] = ""
+		data["authIDRank"] = ""
+		data["authCardHOPLocked"] = FALSE
+		data["trimClockedOut"] = FALSE
+		data["trimAssignment"] = ""
+	return data
+
+/// Places any items inside of the `eligible_items` list to a lockbox, to be opened by the player when they clock back in.
+/mob/living/carbon/human/proc/secure_items(list/eligible_items, obj/incoming_box)
+	var/obj/item/storage/lockbox/timeclock/shame_box = incoming_box
```

Comment:
the timelock lockbox should really be instatiated here and the ID passed in, and this proc doesn't really need to be on human, instead put it wherever you move the other check in and out procs to

Reply:
Done.